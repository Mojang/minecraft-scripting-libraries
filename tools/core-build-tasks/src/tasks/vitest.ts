import { execSync } from 'child_process';

export function vitestTask() {
    return () => {
        execSync('vitest', { stdio: 'inherit' });
    };
}
