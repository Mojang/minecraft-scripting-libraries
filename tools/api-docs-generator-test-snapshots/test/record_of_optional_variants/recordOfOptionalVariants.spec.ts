// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { describe, it } from 'vitest';
import { runGeneratorForTest } from '../runGeneratorForTest';

describe('Record of Optional Variants', () => {
    it('Generates correct output for record of optional variants', () => {
        runGeneratorForTest({
            testDir: __dirname,
            generators: ['ts', 'msdocs'],
        });
    });
});
