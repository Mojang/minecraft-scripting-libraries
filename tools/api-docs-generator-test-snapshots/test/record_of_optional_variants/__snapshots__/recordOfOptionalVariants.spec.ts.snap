// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Record of Optional Variants > Generates correct output for record of optional variants > # Of Files Generated 1`] = `4`;

exports[`Record of Optional Variants > Generates correct output for record of optional variants > msdocs/scriptapi/TOC.yml 1`] = `
"  - name: exampleMember
    items:
    - name: exampleMember
      href: examplemember/examplemember.md
    - name: ClassWithRecordOfOptionalVariantsProperty
      href: examplemember/ClassWithRecordOfOptionalVariantsProperty.md
"
`;

exports[`Record of Optional Variants > Generates correct output for record of optional variants > msdocs/scriptapi/examplemember/ClassWithRecordOfOptionalVariantsProperty.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: examplemember.ClassWithRecordOfOptionalVariantsProperty Class
description: Contents of the exampleMember.ClassWithRecordOfOptionalVariantsProperty class.
---
# ClassWithRecordOfOptionalVariantsProperty Class

## Properties

### **exampleMember**
\`exampleMember?: Record<string, boolean | number | string | undefined>;\`

Type: Record<*string*, *boolean* | *number* | *string*>

Notes:
  - This property can't be used in read-only mode.
"
`;

exports[`Record of Optional Variants > Generates correct output for record of optional variants > msdocs/scriptapi/examplemember/examplemember.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: examplemember Module
description: Contents of the exampleMember module
---
# \`exampleMember\` Module

## Manifest Details
\`\`\`json
{
    "module_name": "exampleMember",
    "version": "1.0.0"
}
\`\`\`
This is version 1.x.x of this module, which is the latest as of version 1.0.0 of Minecraft.

## Available Versions
- \`1.0.0\`

## Classes
- [ClassWithRecordOfOptionalVariantsProperty](ClassWithRecordOfOptionalVariantsProperty.md)
"
`;

exports[`Record of Optional Variants > Generates correct output for record of optional variants > typescript/exampleMember@1.0.0.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "exampleMember",
 *   "version": "1.0.0"
 * }
 * \`\`\`
 *
 */
export class ClassWithRecordOfOptionalVariantsProperty {
    private constructor();
    /**
     * @remarks
     * This property can't be used in read-only mode.
     *
     */
    exampleMember?: Record<string, boolean | number | string | undefined>;
}
"
`;
